[#80] Vite CORS updates and vite.php simplification

joshuapease opened the pull request:
### Overview

- #80 

After some discussion with @maxfenton and @robzor I've made a few edits to our Vite config

- CORS & Allowed host changes. This addresses the [CORS vulnerability that Vite had to patch](https://github.com/vitejs/vite/security/advisories/GHSA-vg6x-rcgg-rjx6).
- Simplified vite.php
    - Previously I'd tried to make the config "just work" on HTTP and HTTPs requests to the site. But it's way cleaner to just assume HTTPS.
- Updated NPM dependencies. 
	- Transitions from Vite 5 → 6. Fortunately, they tried not to have any [breaking changes](https://vite.dev/guide/migration.html#migration-from-v4:~:text=Vite%206%20strives%20to%20avoid%20breaking%20changes%20to%20ensure%20most%20projects%20can%20quickly%20upgrade%20to%20the%20new%20major) for most projects.

### Testing Notes

If anyone else wants to kick the tires.

Create a directory for your new Craft project

```
mkdir my-craft-project
```

#### Then move into your directory
```
cd my-craft-project
```

Run the following composer create-project command.

It uses this PR's branch.

```
docker run --rm -it -v "$PWD":/app -v ${COMPOSER_HOME:-$HOME/.composer}:/tmp composer create-project viget/craft-site-starter=dev-jp/80-vite-config-tweaks-cors-simplify ./ --ignore-platform-reqs
```

After completing the setup steps, run the following commands
```
ddev restart
ddev craft install
```

Configure Craft Multisite & add a new `additional_hostnames` to DDEV

- [ ] Visiting the separate domains should load CSS, JS and fonts without issue.


### Screenshots

**Primary Site**

![CleanShot 2025-02-27 at 15 56 16@2x](https://github.com/user-attachments/assets/85fd8c76-727c-4083-b106-c41dc6cad51b)


** 👯  Multisite example**

![CleanShot 2025-02-27 at 15 57 45@2x](https://github.com/user-attachments/assets/84db6ebb-dc34-4896-a259-2f2bb082f5ce)

joshuapease committed:
[#80] Update npm dependencies

Run `ddev install` with the @latest flag on all dependencies.

joshuapease committed:
[#80] Simplify vite.php

The config can be way simpler if you only allow access from https

joshuapease committed:
[#80] Configure allowed site and cors to work with any DDEV domain.

joshuapease commented on vite.config.js:
```diff
@@ -28,6 +28,10 @@ export default defineConfig(({ command, mode }) => {
       strictPort: true,
       port: HTTP_PORT,
       origin: env.PRIMARY_SITE_URL + ':' + originPort,
+      allowedHosts: ['.ddev.site'],
+      cors: {
+        origin: /https?:\/\/([A-Za-z0-9\-.]+)?(\.ddev\.site)(?::\d+)?$/,
+      },
```
Took me a minute to realize I needed to update BOTH configs.

maxfenton commented on config/vite.php:
```diff
@@ -2,21 +2,10 @@
-    'devServerPublic' => "$host:$devServerPort",
+    // Match to https port in .ddev/config.yaml -> web_extra_exposed_ports
+    // Only works when using HTTPS on DDEV.
+    'devServerPublic' => App::env('PRIMARY_SITE_URL') . ':3001',
```
~Does it make sense to use the variable above to both be more explicit and DRY?~

```suggestion
    'devServerPublic' => App::env('PRIMARY_SITE_URL') . ':3000', // matches ddev web_extra_exposed_ports.https_port
```

maxfenton replied:
Oh wait, now I see you simplified further than I realized. I don't think we should take out the note about the ports needing to match web_extra_exposed_ports. Let me re-read this PR

maxfenton replied:
Here's what I have working on another site:
`.ddev/config.yaml`
```
web_extra_exposed_ports:
  # Expose ports for vite
  - name: vite
    container_port: 3000
    http_port: 3001
    https_port: 3000
```
**note those ports differ from this site**


`config/vite.php`
```
<?php

use craft\helpers\App;

/*
 $host = Craft::$app->getRequest()->getIsConsoleRequest()
    ? App::env('PRIMARY_SITE_URL')
    : Craft::$app->getRequest()->getHostInfo();
*/
$host = App::env('PRIMARY_SITE_URL');

/*
.ddev/config.yaml
    container_port: 3000
    http_port: 3001
    https_port: 3000
 */

$httpPort = 3001;
$httpsPort = 3000;
$devServerPort = str_starts_with($host, 'https') ? $httpsPort : $httpPort;

return [
    'checkDevServer' => true,
    'devServerInternal' => 'http://localhost:' . $httpsPort,
    'devServerPublic' => preg_replace('/:\d+$/', '', App::env('PRIMARY_SITE_URL')) . ':' . $httpsPort,
    'serverPublic' => App::env('PRIMARY_SITE_URL') . '/dist/',
    'useDevServer' => App::env('ENVIRONMENT') === 'dev' || App::env('CRAFT_ENVIRONMENT') === 'dev',
    'manifestPath' => '@webroot/dist/manifest.json', // Vite 4
    'includeReactRefreshShim' => false,
    'includeModulePreloadShim' => false,
    'errorEntry' => '',
    'cacheKeySuffix' => '',
];
```
**now I notice that the final working version doesn't use `$devServerPort` at all**

`vite.config.js`
```
// Match ports in .ddev/config.yaml -> web_extra_exposed_ports
const HTTP_PORT = 3001;
const HTTPS_PORT = 3000;
...
export ...
  return ... 
    server: {
      allowedHosts: ['.ddev.site'],
      host: '0.0.0.0',
      port: HTTPS_PORT,
      strictPort: true,
      origin: `${process.env.DDEV_PRIMARY_URL?.replace(/:\d+$/, '')}:${HTTPS_PORT}`,
      cors: {
        origin: /https?:\/\/([A-Za-z0-9\-\.]+)?(\.ddev\.site)(?::\d+)?$/,
      },
    },
...
```

joshuapease committed:
[#80] PR Feedback

- Add regex replacmenets to allow for custom DDEV ports
- Switch port numbers to match nystudio docs.
  - https://nystudio107.com/docs/vite/#using-ddev
- Use DDEV_PRIMARY_URL in vite.config.js
  - We don't need loadEnv anymore
